src/Microsoft.Health.Fhir.Core/Features/Operations/ConvertData/FhirServerBuilderConvertDataRegistrationExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Operations.ConvertData;
+using Microsoft.Health.Fhir.Core.Registration;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FhirServerBuilderConvertDataRegistrationExtensions
+    {
+        public static IFhirServerBuilder AddConvertData(this IFhirServerBuilder fhirServerBuilder)
+        {
+            EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
+
+            fhirServerBuilder.AddConvertDataTemplateProvider()
+                .AddConvertDataEngine();
+
+            return fhirServerBuilder;
+        }
+
+        private static IFhirServerBuilder AddConvertDataEngine(this IFhirServerBuilder fhirServerBuilder)
+        {
+            fhirServerBuilder.Services.AddSingleton<IConvertDataEngine, ConvertDataEngine>();
+
+            return fhirServerBuilder;
+        }
+
+        private static IFhirServerBuilder AddConvertDataTemplateProvider(this IFhirServerBuilder fhirServerBuilder)
+        {
+            // Add memory cache to store template object and ACR access tokens
+            // ToDo: add size limit
```

Comment:
>// ToDo: add size limit [](start = 12, length = 23)

nit: By size limit do you mean cache size limit? If so I think you already doing this.

Reply:
Fixed. Thanks!